tests/test_report_manager.py

```diff
@@ -242,16 +248,27 @@ def subtest_build_summary_table(self, mode, cpu_only):
             self.report_manager._build_summary_table(
                 report_key="test_model",
                 num_measurements=10,
-                gpu_name="TITAN RTX")
-
-        expected_summary_sentence = (
-            "In 10 measurement(s), config test_model_config_10 (1/GPU model instance(s)"
-            " with max batch size of 8 and dynamic batching enabled) on"
-            " platform tensorflow_graphdef delivers maximum"
-            " throughput under the given constraints")
-        if not cpu_only:
-            expected_summary_sentence += " on GPU(s) TITAN RTX"
-        expected_summary_sentence += "."
+                num_configurations=3,
+                gpu_name="TITAN RTX",
+                cpu_only=cpu_only)
+
+        if cpu_only:
+            expected_summary_sentence = (
+                "In 10 measurements across 3 configurations, "
+                "<strong>test_model_config_10</strong> provides the best throughput: <strong>200 infer/sec</strong>.<br><br>"
+                "This is a <strong>200% gain</strong> over the default configuration "
+                "(100 infer/sec), under the given constraints.<UL><LI> "
+                "<strong>test_model_config_10</strong>: 1/GPU model instances with a max batch size of 8 on platform tensorflow_graphdef "
+                "</LI> </UL>")
+        else:
```

Comment:
Does this test cover all of your refactor and new additions? I think we need more testing here: a high level test like this one for multi-model and/or some lower level tests that hit the lowest level functions to confirm their corner cases.

Reply:
I agree that we need more testing here, but I don't think our current approach gives us the best !/$, it's just not scalable for multi-model. I've filed [TMA-666](https://jirasw.nvidia.com/browse/TMA-666) to add checkpoint based unit testing.

For single model, we have decent coverage with the current unit tests. And, I've got the multi-model correct by inspection (running analysis and looking at the report pdf).